packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -638,20 +618,75 @@ class PaddleSliderValueIndicatorShape extends SliderComponentShape {
     return _bottomLobeEnd;
   }
 
-  void _drawValueIndicator(Canvas canvas, Offset center, Paint paint, double scale,
-      TextPainter labelPainter, double textScaleFactor) {
+  // Determines the "best" offset to keep the bubble on the screen. The calling
+  // code will bound that with the available movement in the paddle shape.
+  double _getIdealOffset(
+    RenderBox parentBox,
+    double halfWidthNeeded,
+    double scale,
+    Offset center,
+    MediaQueryData mediaQueryData,
+  ) {
+    const double edgeMargin = 4.0;
+    final Rect topLobeRect = new Rect.fromLTWH(
+      -_topLobeRadius - halfWidthNeeded,
+      -_topLobeRadius - _distanceBetweenTopBottomCenters,
+      2.0 * (_topLobeRadius + halfWidthNeeded),
+      2.0 * _topLobeRadius,
+    );
+    final Offset origin = center - parentBox.localToGlobal(Offset.zero);
+    // We can just multiply by scale instead of a transform, since we're scaling
+    // around (0, 0).
+    final Offset topLeft = parentBox.localToGlobal((topLobeRect.topLeft * scale) + origin);
+    final Offset bottomRight = parentBox.localToGlobal((topLobeRect.bottomRight * scale) + origin);
+    final Size screenBounds = mediaQueryData?.size ?? (window.physicalSize / window.devicePixelRatio);
```

Comment:
this isn't valid either. You should either use `window` or a media query, but mixing them makes no sense since the whole point of media query is to potentially lie about the window values.

Also if you use window directly you have to subscribe to changes, because otherwise if the user e.g. rotates the phone, you won't necessarily be marked as needing to repaint and your values will be out of date.

Reply:
OK, so I should just require a media query then, right?  I was trying to allow putting the slider into a context where it didn't have a media query available, and the only default (in the absence of a media query) I could think of that made sense was this one.